From 4e3a7dafea7ca3511846027bec8ea3c7d7269754 Mon Sep 17 00:00:00 2001
From: Roman Yepishev <ryepishev@google.com>
Date: Sat, 27 Dec 2025 16:27:15 -0500
Subject: [PATCH] plugins/power: do not unidle for inactive session

NVIDIA drivers change the virtual terminal before suspend and restore it
upon resume.

Upon resume, there is a period of time when systemd reports that our
session is not active (we are on a different virtual terminal). On
resume, `idle_set_mode` would ignore that state change (keeping it as
"sleep"), and if `WakeUpScreen` signal from screensaver arrives at that
time, we will store "sleep" as `previous_idle_mode`, and set up a
callback to restore to that state. That causes the machine to suspend
15 seconds after resume.

This change avoids the issue by not setting up any future action if we
are not in an active session.

Closes #903
---
 plugins/power/gsd-power-manager.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index 1c4dfa692..a4387e497 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -2260,6 +2260,10 @@ set_temporary_unidle_on_ac (GsdPowerManager *manager,
                         manager->temporary_unidle_on_ac_id = 0;
                 }
         } else {
+                /* Don't set up unidle for an inactive session. */
+                if (!manager->session_is_active)
+                        return;
+
                 /* Don't overwrite the previous idle mode when an unidle is
                  * already on-going */
                 if (manager->temporary_unidle_on_ac_id != 0) {
-- 
GitLab
